BDArmory/Weapons/Missiles/MissileLauncher.cs

```diff
@@ -823,6 +842,23 @@ public override float GetBlastRadius()
                     }
                     else
                     {
+                        if (multiLauncher)
+                        {
+                            using (var parts = PartLoader.LoadedPartsList.GetEnumerator())
+                                while (parts.MoveNext())
+                                {
+                                    if (parts.Current.partConfig == null || parts.Current.partPrefab == null)
+                                        continue;
+                                    if (!parts.Current.partPrefab.partInfo.name.Contains(multiLauncher.subMunitionName)) continue;
+                                    foreach (PartModule m in parts.Current.partPrefab.Modules)
+                                    {
+                                        if (m.moduleName == "BDExplosivePart")
+                                        {
+                                            return BlastPhysicsUtils.CalculateBlastRange(m.part.FindModuleImplementing<BDExplosivePart>().GetBlastRadius());
```

Comment:
I see that `GetBlastRadius()` is used in quite a few places that get called often. This and the other if-else branches in `GetBlastRadius` should set `blastRadius` to avoid recalculating it unnecessarily. They'd need to reset it to 0 if the various config options (`tntMass`, Nuke yield, EMP proximity) are changed though.

Reply:
Good catch.